### PR TITLE
prevent saving form if duplicate questions or section headers are used in builder

### PR DESCRIPTION
**Pull request by dodumosu**
see: nditech/apollo-issues#15

### CODE REVIEW

**Review by takinbo — CHANGES_REQUESTED** [review]
Functionality works but notification system doesn't match the UX of the frontend framework. We should use a modal here:
Screenshot from 2021-08-17 05-28-27

**Comment by dodumosu** [comment]
firefox_deCbV7eznH
would this be sufficient?

**Comment by takinbo** [comment]
> would this be sufficient?

Unfortunately not.

Why are we not using [Toasts](https://getbootstrap.com/docs/4.6/components/toasts/)?

There are three solutions I can suggest:

1. Toasts (Bootstrap)
2. Alerts (Bootstrap)
3. A simple modal dialog with a close button

**Comment by takinbo** [comment]
We're almost there. Can you split this into two message alerts. One for the sections and another for the question codes. I've made prototypes:

Screenshot from 2021-08-20 17-44-25

Screenshot from 2021-08-20 17-42-19

**Review by takinbo — CHANGES_REQUESTED** [review]
See general comments

**Comment by takinbo** [comment]
Also, when you're done, please squash your commits into one.

File: apollo/templates/admin/formbuilder.html [hunk 1]
```diff
@@ -2,6 +2,7 @@
 {% block head_css %}
 {{ super() }}
 <link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='css/angular-form-builder.css') }}">
+<link rel="stylesheet" href="{{ url_for('pwa.static', filename='vendor/notiflix/notiflix-2.7.0.min.css') }}">
```
Comment (takinbo) [inline apollo/templates/admin/formbuilder.html]: Not necessary since it's not being used.

File: apollo/templates/admin/formbuilder.html [hunk 2]
```diff
@@ -62,12 +77,13 @@ <h5 class="card-title mb-0">{{ form.name }}</h5>
 {% block tail_js %}
 {{ super() }}
 <script type="text/javascript" src="{{ asset_url_for('gettext.js') }}" charset="utf-8"></script>
-<script src="{{ url_for('static', filename='js/simply-toast.min.js') }}"></script>
+<script src="{{ url_for('pwa.static', filename='vendor/notiflix/notiflix-2.7.0.min.js') }}"></script>
```
Comment (takinbo) [inline apollo/templates/admin/formbuilder.html]: Should be removed since it's not being used

**Review by takinbo — CHANGES_REQUESTED** [review]

**Review by takinbo — APPROVED** [review]
tACK

**Pull request merged**